akka-actor-typed-tests/src/test/java/jdocs/akka/typed/MutableIntroTest.java

```diff
@@ -138,4 +140,53 @@ public ChatRoomBehavior(ActorContext<RoomCommand> ctx) {
   }
   //#chatroom-actor
 
+
+  //#chatroom-gabbler
+  public static abstract class Gabbler {
+    private Gabbler() {
+    }
+
+    public static Behavior<ChatRoom.SessionEvent> behavior() {
+      return Behaviors.receive(ChatRoom.SessionEvent.class)
+          .onMessage(ChatRoom.SessionDenied.class, (ctx, msg) -> {
+            System.out.println("cannot start chat room session: " + msg.reason);
+            return Behaviors.stopped();
+          })
+          .onMessage(ChatRoom.SessionGranted.class, (ctx, msg) -> {
+            msg.handle.tell(new ChatRoom.PostMessage("Hello World!"));
+            return Behaviors.same();
+          })
+          .onMessage(ChatRoom.MessagePosted.class, (ctx, msg) -> {
+            System.out.println("message has been posted by '" +
+                msg.screenName +"': " + msg.message);
+            return Behaviors.stopped();
+          })
+          .build();
+    }
+
+  }
+  //#chatroom-gabbler
+
+  public static void runChatRoom() throws Exception {
+
+    //#chatroom-main
+    Behavior<Void> main = Behaviors.setup(ctx -> {
+      ActorRef<ChatRoom.RoomCommand> chatRoom =
+          ctx.spawn(ChatRoom.behavior(), "chatRoom");
+      ActorRef<ChatRoom.SessionEvent> gabbler =
+          ctx.spawn(Gabbler.behavior(), "gabbler");
+      ctx.watch(gabbler);
+      chatRoom.tell(new ChatRoom.GetSession("ol’ Gabbler", gabbler));
+
+      return Behaviors.receive(Void.class)
+          .onSignal(Terminated.class, (c, sig) -> Behaviors.stopped())
+          .build();
+    });
+
+    final ActorSystem<Void> system =
+        ActorSystem.create(main, "ChatRoomDemo");
+
+    system.getWhenTerminated().toCompletableFuture().get();
```

Comment:
that is not needed, the process will stay alive as long as the ActorSystem is running

Reply:
Yeah, I was thinking that when I copy-pasted, I'll remove it in both samples.